plugins/inputs/intel_dlb/intel_dlb.go

```diff
@@ -0,0 +1,470 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+// +build linux
+
+package intel_dlb
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type IntelDLB struct {
+	SocketPath       string          `toml:"socket_path"`
+	EventdevCommands []string        `toml:"eventdev_commands"`
+	DLBDeviceIDs     []string        `toml:"dlb_device_types"`
+	Log              telegraf.Logger `toml:"-"`
+
+	connection           net.Conn
+	devicesDir           []string
+	rasReader            rasReader
+	maxInitMessageLength uint32
+	logOnce              map[string]error
+}
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//
+//go:embed sample.conf
+var sampleConfig string
+
+const (
+	defaultSocketPath      = "/var/run/dpdk/rte/dpdk_telemetry.v2"
+	pluginName             = "intel_dlb"
+	eventdevListCommand    = "/eventdev/dev_list"
+	dlbDeviceIDLocation    = "/sys/devices/*/*/device"
+	aerCorrectableFileName = "aer_dev_correctable"
+	aerFatalFileName       = "aer_dev_fatal"
+	aerNonFatalFileName    = "aer_dev_nonfatal"
+	defaultDLBDevice       = "0x2710"
+)
+
+// SampleConfig returns sample config
+func (d *IntelDLB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init performs validation of all parameters from configuration.
+func (d *IntelDLB) Init() error {
+	var err error
+
+	if d.SocketPath == "" {
+		d.SocketPath = defaultSocketPath
+		d.Log.Debugf("Using default '%v' path for socket_path", defaultSocketPath)
+	}
+
+	err = checkSocketPath(d.SocketPath)
+	d.logErrorOnce("socketPathDoesNotExist", err)
+
+	if len(d.EventdevCommands) == 0 {
+		eventdevDefaultCommands := []string{"/eventdev/dev_xstats", "/eventdev/port_xstats", "/eventdev/queue_xstats", "/eventdev/queue_links"}
+		d.EventdevCommands = eventdevDefaultCommands
+		d.Log.Debugf("Using default eventdev commands '%v'", eventdevDefaultCommands)
+	}
+
+	if err = validateEventdevCommands(d.EventdevCommands); err != nil {
+		return err
+	}
+
+	if len(d.DLBDeviceIDs) == 0 {
+		d.DLBDeviceIDs = []string{defaultDLBDevice}
+		d.Log.Debugf("Using default DLB Device ID '%v'", defaultDLBDevice)
+	}
```

Comment:
Similar to Hipsak's question, what is this default device? Will everyone have it? Or should this be empty and require the user to specify a device? 

Is this device ID like a PCI ID? What happens if a device ID exists multiple times?

Reply:
Right now, only supported and tested device id is `0x2710`. Configuration field was added to support forward compatibility.

Proper information added to `sample.conf` and `README.md`.

To have this default device, user need to follow instruction from User Guide which can be found here: https://www.intel.com/content/www/us/en/download/686372/intel-dynamic-load-balancer.html (also mentioned in README).